### PR TITLE
Fix potential bugs

### LINKED ISSUE #1: TypeError: 'set' on proxy: trap returned falsish for property 'foo'
Hi,

I tried to use `defaultdict2` to set a value but it throws the error:

```bash
TypeError: 'set' on proxy: trap returned falsish for property 'foo'
```

I think https://github.com/wilzbach/defaultdict2/blob/master/src/index.js#L50 must `return true` at the end of `set` handler.

### PR DESCRIPTION
**Pull request by vunb**
* Fix #1 
* Default create new object without any properties

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -53,9 +53,10 @@ var defaultdict = function(){
                 }else{
                     obj[prop] = value;
                 }
+                return true;
```
Comment (wilzbach) [inline src/index.js:56]: FWIW the documentation doesn't mention this (e.g. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set) and on my Chrome & node.js this works fine without `return`. Out of interest: what browser or engine do you use?

**Review by wilzbach — COMMENTED** [review]
I should have definitely added some tests to this :/

File: src/index.js (hunk 1 above)
Comment (vunb) [reply src/index.js:56]: I tried with nodejs version `6.9.1`.

Comment (vunb) [reply src/index.js:56]: Hi @wilzbach 

Acording to this document https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/handler/set

> The set method should return a boolean value. Return true to indicate that assignment succeeded. If the set method returns false, and the assignment happened in strict-mode code, a TypeError will be thrown.

In my code, i `use strict`, then a TypeError was fired.

Please consider this case!

Thanks.

File: index.js [hunk 2]
```diff
@@ -6,7 +6,8 @@ try{
     Proxy.create({}, handler);
 }catch(e){
     // hack to enable the harmony-proxies flag by default
-    require("harmonize")(["harmony-proxies"]);
+    require("harmonize")(["harmony", "harmony-proxies", "harmony_proxies"]);
+    require('harmony-proxy');
```
Comment (wilzbach) [inline index.js]: What's the motivation for using both packages?

Comment (vunb) [reply index.js]: oh, this is my bad attempt. it is only need harmonize@2.0.0
It works after update 👍 
https://travis-ci.org/vntk/vntk-conlleval/builds/323469196

**Pull request closed without merging**